arbnode/batch_poster.go

```diff
@@ -194,7 +200,9 @@ var DefaultBatchPosterConfig = BatchPosterConfig{
 	Enable:                             false,
 	DisableDasFallbackStoreDataOnChain: false,
 	// This default is overridden for L3 chains in applyChainParameters in cmd/nitro/nitro.go
-	MaxSize:            100000,
+	MaxSize: 100000,
+	// TODO: is 1000 bytes an appropriate margin for error vs blob space efficiency?
+	Max4844BatchSize:   (254 * params.BlobTxFieldElementsPerBlob / 8 * (params.MaxBlobGasPerBlock / params.BlobTxBlobGasPerBlob)) - 1000,
```

Comment:
Is there some way a batch that is `(254 * params.BlobTxFieldElementsPerBlob / 8 * (params.MaxBlobGasPerBlock / params.BlobTxBlobGasPerBlob))` could expand to be bigger than `(256 * params.BlobTxFieldElementsPerBlob / 8 * (params.MaxBlobGasPerBlock / params.BlobTxBlobGasPerBlob))` bytes when encoded as field elements? In any case it's probably worth being cautious about at the beginning.

Reply:
No, this math is based on us being able to pack 254 bits into every field element guaranteed. The 1000 byte margin of error is for compression and that we might exceed the max size we're aiming for (I'm unsure if this is still possible or it was a bug that we already fixed).